### PR TITLE
sensors: add support for icm42688 interrupt

### PR DESCRIPTION
**Pull request by yperess**

### CODE REVIEW

**Review by teburd — DISMISSED** [review]
This looks fine to me

File: drivers/sensor/icm42688/icm42688_emul.c [hunk 1]
```diff
@@ -41,6 +41,20 @@ void icm42688_emul_get_reg(const struct emul *target, uint8_t reg_addr, uint8_t
...
 
+static void icm42688_emul_handle_write(const struct emul *target, uint8_t regn, uint8_t value)
+{
+	struct icm42688_emul_data *data = target->data;
+
+	switch (regn) {
+	case REG_DEVICE_CONFIG:
+		if (FIELD_GET(BIT_SOFT_RESET, value) == 1) {
+			/* Perform a soft reset */
+			data->reg[REG_INT_STATUS] |= BIT_INT_STATUS_RESET_DONE;
```
Comment (tristan-google) [inline drivers/sensor/icm42688/icm42688_emul.c:52]: Is there any internal state that needs resetting?

File: samples/sensor/sensor_shell/src/main.c [hunk 2]
```diff
@@ -4,6 +4,64 @@
...
+	if (sensor_sample_fetch(sensor)) {
+		LOG_ERR("Failed to fetch samples on data ready handler");
+	}
+	for (int i = 0; i < SENSOR_CHAN_ALL; ++i) {
+		/* Skip 3 axis channels */
+		if (i == SENSOR_CHAN_ACCEL_XYZ || i == SENSOR_CHAN_GYRO_XYZ ||
+		    i == SENSOR_CHAN_MAGN_XYZ) {
+			continue;
+		}
+		if (sensor_channel_get(sensor, i, &value) != 0) {
```
Comment (MaureenHelm) [inline samples/sensor/sensor_shell/src/main.c:36]: This causes a lot of errors on `frdm_k64f` when you enable the data ready trigger:
```
$ west build -b frdm_k64f -p auto samples/sensor/sensor_shell/ -- -DCONFIG_FXOS8700_TRIGGER_GLOBAL_THREAD=y
```
```console
--- 30 messages dropped ---
[00:00:09.459,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.459,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.459,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
--- 29 messages dropped ---
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.622,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
[00:00:09.785,000] <err> FXOS8700: Unsupported sensor channel
```

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/sensor/icm42688/icm42688_emul.c (hunk 1 above)
Comment (yperess) [reply drivers/sensor/icm42688/icm42688_emul.c:52]: Done

File: samples/sensor/sensor_shell/src/main.c (hunk 2 above)
Comment (yperess) [reply samples/sensor/sensor_shell/src/main.c:36]: I updated the logic to try once and if we get `-ENOTSUP` then never try that channel again

Comment (teburd) [reply samples/sensor/sensor_shell/src/main.c]: Right we don't really have a nice way of introspecting supported channels at runtime or compile time at the moment per sensor. Makes sense to me to try once.

**Review by teburd — APPROVED** [review]
Makes sense to me

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**